### ISSUE TITLE
Simplified installation under OSX with Homebrew tap

### ISSUE THREAD

**Issue by awendt** [entropia/tip-toi-reveng#95]
I'm running OSX and wanted to try out `tttool`. I did not want to install manually, so I wrote a formula for [Homebrew](https://github.com/Homebrew/homebrew/), the de facto package manager for OSX that I'm using anyway.

When you have Homebrew installed, you can simply install `tttool` like this:

```
brew tap awendt/tap
brew install tttool
```

Dependencies like haskell and Cabal are resolved by Homebrew using [the formula](https://github.com/awendt/homebrew-tap/blob/master/tttool.rb), and the installation is cleanly sandboxed.

I'm not sure it's a good idea to sanction [my repo](https://github.com/awendt/homebrew-tap) as the "official tap" (you'd want things like that under your control) but I'd like to see a tap available nonetheless.

Thoughts on this? Would you be willing to create a formulae repo for entropia?

**Comment by nomeata** [comment]
I am completely ignorant about OSX-issues and I’m happy to either sanction your repo as a formulae repository, or have one in the entropia namespace (which I would have to coordinate with them).

Does `tap` only support GitHub repositories, or would it also be possible and helpful to put the formula into a regular webspace, e.g. `tttool.entropia.de`?

**Comment by awendt** [comment]
> Does tap only support GitHub repositories, or would it also be possible and 
> helpful to put the formula into a regular webspace, e.g. tttool.entropia.de?

Short answer: No.

> **tl;dr** `brew tap` allows you to add more GitHub repos to the list of formulae that brew tracks,
> updates and installs from.
> — [docs on brew tap](https://github.com/Homebrew/homebrew/blob/d93e5d16f27d2aa5fe7dc3f8c782c1adea33c174/share/doc/homebrew/brew-tap.md)

Apart from (1) my own repo as source and (2) one under entropia namespace, there is a third solution: `brew install` [supports specifying formulae as URL](https://github.com/Homebrew/homebrew/blob/3d78f641ede359071e0695e2499bf6b29076dadb/Library/Homebrew/manpages/brew.1.md#specifying-formulae).

We could put my formula under version control here. This seems easier but more hacky. For one, I'm not sure how updates of the formula file are handled by `brew install` (we might just have to find out...)

**Comment by nomeata** [comment]
What are the requirements for setting up a tap in the entropia namespace? Is it just any git repository named homebrew-*foo*? E.g. would "github.com/entropia/homebrew-tttool" work?

**Comment by awendt** [comment]
Yes, that would work:

> On GitHub, your repository must be named homebrew-something. The prefix 'homebrew-' is not optional.

[src](https://github.com/Homebrew/homebrew/blob/d93e5d16f27d2aa5fe7dc3f8c782c1adea33c174/share/doc/homebrew/brew-tap.md)

**Comment by nomeata** [comment]
Ok, created https://github.com/entropia/homebrew-tttool and added that ruby file. Did I do it correctly? Would you mind creating a pull request with a README, and also submit a corresponding change to this repository’s README and gh-pages (which produce tttool.entropia.de)?

As I don’t have an OS X system, I hope you’ll stick around to maintain the homebrew repo. Can you maybe add some contact information to the repo and/or formula file?

**Comment by nomeata** [comment]
@awendt: Did you have a chance to look at the homebrew-tttool repo I created?

**Comment by awendt** [comment]
@nomeata Sorry for taking so long. I could successfully install `tttool` using that repo, thanks. I'll send a PR for the latest release (1.4.x) soon.
